internal/venice-common/src/main/java/com/linkedin/venice/system/store/MetaStoreWriter.java

```diff
@@ -371,16 +372,61 @@ private void update(
     }
     StoreMetaKey key = dataType.getStoreMetaKey(keyStringSupplier.get());
     SpecificRecord update = updateSupplier.get();
-    writer.update(
-        key,
-        update,
-        AvroProtocolDefinition.METADATA_SYSTEM_SCHEMA_STORE.currentProtocolVersion.get(),
-        derivedComputeSchemaId,
-        null);
-    writer.flush();
+    writeMessageWithRetry(metaStoreName, vw -> {
+      vw.update(
+          key,
+          update,
+          AvroProtocolDefinition.METADATA_SYSTEM_SCHEMA_STORE.currentProtocolVersion.get(),
+          derivedComputeSchemaId,
+          null);
+    });
+  }
+
+  void writeMessageWithRetry(String metaStoreName, Consumer<VeniceWriter> writerConsumer) {
+    ReentrantLock lock = getOrCreateMetaStoreWriterLock(metaStoreName);
+    int messageProduceRetryCount = 0;
+    int maxMessageProduceRetryCount = 3;
+    VeniceWriter writer = null;
+    boolean messageProduced = false;
+    while (!messageProduced) {
+      try {
+        lock.lock();
```

Comment:
Not a big issue, just try to align with the existing code style.
Can you move this line up right above `try`?

Reply:
Sure, just updated.